bokeh/core/properties.py

```diff
@@ -711,6 +711,38 @@ def __setattr__(self, name, value):
             raise AttributeError("unexpected attribute '%s' to %s, %s attributes are %s" %
                 (name, self.__class__.__name__, text, nice_join(matches)))
 
+    # this is like _value_record_references but expects to find refs
+    # instead of models, and takes a doc to look up the refs in
+    def _json_record_references(doc, v, result, recurse):
+        if v is None: return
+        if isinstance(v, dict) and set(v.keys()) == set(['id', 'type']):
```

Comment:
there's an optional `subtype` key (see `def ref`) I guess, so it should be that the set is a superset of `['id', 'type']`